daemon/src/rollover_maker.rs

```diff
@@ -0,0 +1,246 @@
+use crate::maker_cfd::RollOverCompleted;
+use crate::maker_inc_connections::TakerMessage;
+use crate::model::cfd::{OrderId, Role};
+use crate::model::Identity;
+use crate::oracle::GetAnnouncement;
+use crate::projection::UpdateRollOverProposal;
+use crate::setup_contract::RolloverParams;
+use crate::tokio_ext::spawn_fallible;
+use crate::wire::{MakerToTaker, RollOverMsg};
+use crate::{
+    log_error, maker_inc_connections, oracle, projection, schnorrsig, setup_contract, wire, Cfd,
+    Stopping,
+};
+use anyhow::{Context as _, Result};
+use async_trait::async_trait;
+use futures::channel::mpsc;
+use futures::channel::mpsc::UnboundedSender;
+use futures::{future, SinkExt};
+use xtra::prelude::MessageChannel;
+use xtra::{Context, Handler, KeepRunning};
+
+pub struct AcceptRollOver;
+
+pub struct RejectRollOver {
+    pub order_id: OrderId,
+}
+
+pub struct Actor {
+    send_to_taker_actor: Box<dyn MessageChannel<TakerMessage>>,
+    cfd: Cfd,
+    taker_id: Identity,
+    n_payouts: usize,
+    oracle_pk: schnorrsig::PublicKey,
+    sent_from_taker: Option<UnboundedSender<RollOverMsg>>,
+    maker_cfd_actor: Box<dyn MessageChannel<RollOverCompleted>>,
+    oracle_actor: Box<dyn MessageChannel<GetAnnouncement>>,
+    on_stopping: Vec<Box<dyn MessageChannel<Stopping<Self>>>>,
+    projection_actor: xtra::Address<projection::Actor>,
+}
+
+#[async_trait::async_trait]
+impl xtra::Actor for Actor {
+    async fn stopping(&mut self, ctx: &mut Context<Self>) -> KeepRunning {
+        let address = ctx.address().expect("acquired own actor address");
+
+        for channel in self.on_stopping.iter() {
+            let _ = channel
+                .send(Stopping {
+                    me: address.clone(),
+                })
+                .await;
+        }
+
+        KeepRunning::StopAll
+    }
+}
+
+impl Actor {
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        send_to_taker_actor: &(impl MessageChannel<TakerMessage> + 'static),
+        cfd: Cfd,
+        taker_id: Identity,
+        oracle_pk: schnorrsig::PublicKey,
+        maker_cfd_actor: &(impl MessageChannel<RollOverCompleted> + 'static),
+        oracle_actor: &(impl MessageChannel<GetAnnouncement> + 'static),
+        (on_stopping0, on_stopping1): (
+            &(impl MessageChannel<Stopping<Self>> + 'static),
+            &(impl MessageChannel<Stopping<Self>> + 'static),
+        ),
+        projection_actor: xtra::Address<projection::Actor>,
+    ) -> Self {
+        Self {
+            send_to_taker_actor: send_to_taker_actor.clone_channel(),
+            cfd,
+            taker_id,
+            n_payouts: 0,
```

Comment:
Uhm, probably better not :D

Reply:
are you talking about `n_payouts?`

Reply:
Yes, we should not set to zero :D

Reply:
ok using the constant (200)

Reply:
I think we're meant to use the field in `maker_cfd::Actor`: https://github.com/itchysats/itchysats/blob/de36ff69028fe1043ae39e68406693b1c5533312/daemon/src/maker_cfd.rs#L96

Reply:
Yes, you are meant to pass it on from the `maker_cfd::Actor`!